src/mod_mam_utils.erl

```diff
@@ -207,16 +207,31 @@ encode_compact_uuid(Microseconds, NodeId)
 
 %% @doc Extract date and node id from a message id.
 -spec decode_compact_uuid(integer()) -> {integer(), byte()}.
-decode_compact_uuid(Id) ->
+decode_compact_uuid(Id) when is_integer(Id) ->
     Microseconds = Id bsr 8,
     NodeId = Id band 255,
-    {Microseconds, NodeId}.
+    {Microseconds, NodeId};
+decode_compact_uuid(UUID) when is_binary(UUID) ->
```

Comment:
At _least_ this should be covered with tests.

Reply:
Where should that go? Would eunit be ok or just ct?

Reply:
I guess into big tests mam_SUITE.
Because we already have tests for all other MAM features there.